t/email.t

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env perl
+
+use 5.028;
+use warnings;
+use experimental qw(signatures);
+
+use Test::Lib;
+use Test::PromAlertProxy;
+use Test::More;
+use Test::Deep;
+use Test::Time time => Test::PromAlertProxy->now;
+
+use IO::Async::Loop;
+
+use PromAlertProxy::Hub;
+use PromAlertProxy::PromAlert;
+
+use PromAlertProxy::Logger '$Logger' => { init => {
+  ident    => 'promalertproxy',
+  to_self  => 1,
+} };
+
+use PromAlertProxy::Target::Email;
+
+my $loop = IO::Async::Loop->new;
+
+my $hub = PromAlertProxy::Hub->new(loop => $loop);
+
+my $target = PromAlertProxy::Target::Email->new(
+  hub       => $hub,
+  id        => 'email',
+  default   => 1,
+  from      => 'from@localhost',
+  to        => 'to@localhost',
+  transport => 'Test',
+);
+$hub->add_target($target);
+
+my %alert_contents = Test::PromAlertProxy->prom_alert->%*;
+my $alert = PromAlertProxy::PromAlert->new(%alert_contents);
+$hub->dispatch($alert);
+
+my $mail_transport = $target->_transport;
+
+is($mail_transport->delivery_count, 1, 'email alert receieved');
+
+my $email = $mail_transport->shift_deliveries;
+if ($email) {
```

Comment:
(or `if (my $email = $transport->shift_deliveries)`, but it hardly matters.)

Reply:
From the "huh, I normally would do it that way, why didn't I" department.